addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScannerUnitTest.java

```diff
@@ -151,35 +202,118 @@ public void shouldAlertOnWildcardFrameAncestorsDirective() throws HttpMalformedH
+    public void shouldNotAlertOnReasonableCsp() {
         // Given
-        HttpMessage msg = new HttpMessage();
-        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
-
-        msg.setResponseBody("<html></html>");
-        msg.setResponseHeader(
-                "HTTP/1.1 200 OK\r\n"
-                        + "Server: Apache-Coyote/1.1\r\n"
-                        + "Content-Security-Policy: default-src 'self'; script-src 'self' "
-                        + "storage.googleapis.com cdn.temasys.io cdn.tiny.cloud *.google-analytics.com; "
-                        + "style-src 'self' *.googleapis.com; font-src 'self' data: *.googleapis.com "
-                        + "fonts.gstatic.com; frame-ancestors 'none'; worker-src 'self'\r\n"
-                        + "Content-Type: text/html;charset=ISO-8859-1\r\n"
-                        + "Content-Length: "
-                        + msg.getResponseBody().length()
-                        + "\r\n");
+        HttpMessage msg = createHttpMessageWithReasonableCsp(HTTP_HEADER_CSP);
         // When
         scanHttpResponseReceive(msg);
         // Then
         assertThat(alertsRaised.size(), equalTo(0));
     }
+
+    @Test
+    public void shouldRaiseAlertOnCspIncludedViaXCSP() {
+        // Given
+        HttpMessage msg = createHttpMessageWithReasonableCsp("X-Content-Security-Policy");
+        // When
+        scanHttpResponseReceive(msg);
+        // Then
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(
+                alertsRaised.get(0).getName(), equalTo("CSP Scanner: X-Content-Security-Policy"));
+        assertThat(alertsRaised.get(0).getRisk(), equalTo(Alert.RISK_LOW));
+        assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
+    }
+
+    @Test
+    public void shouldRaiseAlertOnCspIncludedViaXWKCSP() {
+        // Given
+        HttpMessage msg = createHttpMessageWithReasonableCsp("X-WebKit-CSP");
+        // When
+        scanHttpResponseReceive(msg);
+        // Then
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getName(), equalTo("CSP Scanner: X-WebKit-CSP"));
+        assertThat(alertsRaised.get(0).getRisk(), equalTo(Alert.RISK_LOW));
+        assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
+    }
+
+    @Test
+    public void shouldRaiseAlertWhenCspIncludesScriptUnsafeInline() {
+        // Given
+        HttpMessage msg = createHttpMessage("", "script-src 'unsafe-inline'");
+        // When
+        scanHttpResponseReceive(msg);
+        // Then
+        assertThat(alertsRaised.size(), equalTo(2));
+
+        assertThat(alertsRaised.get(1).getName(), equalTo("CSP Scanner: script-src unsafe-inline"));
+        assertThat(alertsRaised.get(0).getRisk(), equalTo(Alert.RISK_MEDIUM));
+        assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
```

Comment:
Were these really meant to assert the first alert? (Same comment for following test.)